cg/apps/downsample/downsample.py

```diff
@@ -0,0 +1,169 @@
+        """
+        try:
+            return DownsampleData(
+                status_db=self.status_db,
+                hk_api=self.housekeeper_api,
+                sample_id=self.sample_id,
+                number_of_reads=self.number_of_reads,
+                case_id=self.case_id,
+                out_dir=Path(self.config.downsample),
+            )
+        except Exception as error:
+            raise DownsampleFailedError(repr(error))
+
+    def add_downsampled_sample_entry_to_statusdb(self) -> Sample:
+        """Add a down sampled sample entry to StatusDB."""
+        downsampled_sample: Sample = self.downsample_data.downsampled_sample
+        if sample_exists_in_statusdb(
+            status_db=self.status_db, sample_id=downsampled_sample.internal_id
+        ):
+            raise ValueError(f"Sample {downsampled_sample.internal_id} already exists in StatusDB.")
+        LOG.info(
+            f"New downsampled sample created: {downsampled_sample.internal_id} from {downsampled_sample.from_sample}"
+            f"Application tag set to: {downsampled_sample.application_version.application.tag}"
+            f"Customer set to: {downsampled_sample.customer}"
+        )
+        if not self.dry_run:
+            self.status_db.session.add(downsampled_sample)
+            self.status_db.session.commit()
+            LOG.info(f"Added {downsampled_sample.name} to StatusDB.")
+            return downsampled_sample
+        return downsampled_sample
+
+    def add_downsampled_case_to_statusdb(self) -> Family | None:
+        """
+        Add a down sampled case entry to StatusDB.
+        """
+        downsampled_case: Family = self.downsample_data.downsampled_case
+        if case_exists_in_statusdb(status_db=self.status_db, case_name=downsampled_case.name):
+            LOG.info(f"Case with name {downsampled_case.name} already exists in StatusDB.")
+            return
+        if not self.dry_run:
+            self.status_db.session.add(downsampled_case)
+            self.status_db.session.commit()
+            LOG.info(f"New down sampled case created: {downsampled_case.internal_id}")
+            return downsampled_case
+        return downsampled_case
+
+    def _link_downsampled_sample_to_case(self, sample: Sample, case: Family) -> None:
+        """Create a link between sample and case in statusDB."""
+        if self.dry_run:
+            LOG.info(
+                f"Would relate sample {sample} to case {case.internal_id} with name {case.name}"
+            )
+            return
+
+        sample_case_link: FamilySample = self.status_db.relate_sample(
+            family=case,
+            sample=sample,
+            status=self.downsample_data.sample_status(sample=sample),
+        )
+        self.status_db.session.add(sample_case_link)
+        self.status_db.session.commit()
+        LOG.info(f"Related sample {sample.internal_id} to {case.internal_id}")
+
+    def add_downsampled_sample_case_to_statusdb(self) -> None:
```

Comment:
Some naming ideas, I think `add_downsampled_case_to_statusdb` rolls better off the tongue or `store_downsampled_case`. Does not really matter, but if you like it better 👍